benchmark/benchmarks.jl

```diff
@@ -40,51 +51,58 @@ for subdir ∈ subdirs
       push!(names_sqd, name)
       println(name)
 
-      p_amd      = amd(A)
-      p_symamd   = symamd(A)
-      p_metis, _ = Metis.permutation(A)
-      p_symrcm   = symrcm(A)
-      p_classic  = collect(1:size(A,1))
-
-      ldlt_amd     = ldl(A, p_amd)
-      ldlt_symamd  = ldl(A, p_symamd)
-      ldlt_metis   = ldl(A, p_metis)
-      ldlt_symrcm  = ldl(A, p_symrcm)
-      ldlt_classic = ldl(A, p_classic)
-
-      push!(nnz_sqd, nnz_A)
-      push!(ratio_amd, nnz(ldlt_amd) / nnz_A)
-      push!(ratio_symamd, nnz(ldlt_symamd) / nnz_A)
-      push!(ratio_metis, nnz(ldlt_metis) / nnz_A)
-      push!(ratio_symrcm, nnz(ldlt_symrcm) / nnz_A)
-      push!(ratio_classic, nnz(ldlt_classic) / nnz_A)
+      # p_amd      = amd(A)
+      # p_symamd   = symamd(A)
+      # p_metis, _ = Metis.permutation(A)
+      # p_symrcm   = symrcm(A)
+      # p_classic  = collect(1:size(A,1))
+
+      # ldlt_amd     = ldl(A, p_amd)
+      # ldlt_symamd  = ldl(A, p_symamd)
+      # ldlt_metis   = ldl(A, p_metis)
+      # ldlt_symrcm  = ldl(A, p_symrcm)
+      # ldlt_classic = ldl(A, p_classic)
+
+      # push!(nnz_sqd, nnz_A)
+      # push!(ratio_amd, nnz(ldlt_amd) / nnz_A)
+      # push!(ratio_symamd, nnz(ldlt_symamd) / nnz_A)
+      # push!(ratio_metis, nnz(ldlt_metis) / nnz_A)
+      # push!(ratio_symrcm, nnz(ldlt_symrcm) / nnz_A)
+      # push!(ratio_classic, nnz(ldlt_classic) / nnz_A)
+
+      SUITE["no_ordering"][name] = @benchmarkable collect(1:size($A,1))
```

Comment:
What you want here is to perform the factorization with ordering `1:n` and count the number of nonzero elements in the L factor, and then the ratio `nnz(L) / nnz(tril(A, -1))`.

Reply:
Do you mean nnz(tril(A)) instead of nnz(tril(A, -1))?

Reply:
Well, [this line](https://github.com/JuliaSmoothOptimizers/AMD.jl/pull/38/files#diff-d10aa9100c933690e3563ac8c2aef6884efd2a930f2edb898af61f409de12d9aR48) used to say `nnz(tril(A))` but I think it should be `nnz(tril(A, -1))` because we don't store the diagonal of the L factor. So we want to compare the number of nonzeros in L (which is strictly lower triangular) to the number of nonzeros in the strict lower triangle of A.

Reply:
`A` is the matrix that we want to factorize so I suppose that it's relevant to count the nnz elements on its diagonal when `A` is symmetric.
With the LDL' factorization, L have a unit diagonal so we don't count the ones but we also have the elements of D.
When I did the benchmarks, I used the ratio `nnz(tril(L, -1)) + nnz(D) / nnz(tril(A))`.

Reply:
D must be full and diag(A) must be full for the factorization to exist. So I think it's sufficient to compare nnz(L) to nnz(tril(A, -1)). The diagonal of L is not stored.